src/index.d.ts

```diff
@@ -1,115 +1,151 @@
+   * Otherwise, the callback is invoked once the first agent data is received from upstream.
+   * This callback is provided with an `Agent` API object, which can also be created at any time after initialization is complete via `new connect.Agent()`.
    *
-   * A callback to receive contact details
+   * @param callback A callback that will receive an `Agent` API object instance.
+   */
+  function agent(callback: AgentCallback): void;
+
+  /**
+   * A callback to receive `Contact` API object instances.
    *
-   * @param agent A Contact object containing information about the current contact.
+   * @param contact A `Contact` API object instance.
    */
   type ContactCallback = (contact: Contact) => void;
 
   /**
-   * Register a callback to receive contact details
+   * Subscribe a method to be called for each newly detected agent contact.
+   * Note that this function is not only for incoming contacts, but for contacts which already existed when Streams was initialized, such as from a previous agent session.
+   * This callback is provided with a `Contact` API object for this contact. `Contact` API objects can also be listed from the `Agent` API by calling `agent.getContacts()`.
    *
-   * @param callback A callback that will receive a Contact instance
-   *                 when contact information is updated.
+   * @param callback A callback that will receive an `Contact` API object instance.
    */
   function contact(callback: ContactCallback): void;
 
   /**
-   * Binds the given instance object as the context for
-   * the method provided.
+   * A useful utility function for creating callback closures that bind a function to an object instance.
    *
-   * @param scope The instance object to be set as the scope
-   *              of the function.
+   * @param scope The instance object to be set as the scope of the function.
    * @param method The method to be encapsulated.
-   *
-   * All other arguments, if any, are bound to the method
-   * invocation inside the closure.
-   *
-   * @return A closure encapsulating the invocation of the
-   *    method provided in context of the given instance.
    */
-  function hitch(scope: object, method: () => any): void;
+  function hitch<T extends Function>(scope: object, method: T): T;
 
   interface Core {
-    initCCP(containerDiv: HTMLElement, options: InitCCPOptions): void;
+    /**
+     * Integrates with Amazon Connect by loading the pre-built CCP located at `ccpUrl` into an iframe and placing it into the `containerDiv` provided.
+     * API requests are funneled through this CCP and agent and contact updates are published through it and made available to your JS client code.
+     *
+     * @param containerDiv The `<div>` DOM element to place the CCP.
+     * @param options The CCP init options.
+     */
+    initCCP(containerDiv: HTMLDivElement, options: InitCCPOptions): void;
```

Comment:
I don't believe a `div` is required, it can be any container element (as long as it can render the iframe inside of it). I think `HTMLElement` is better here.

Reply:
OK. I decided to update it to `HTMLDivElement` since the argument name was `containerDiv`. Will update with a more lax type.